lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlobFileSystem.cs

```diff
@@ -0,0 +1,255 @@
+    /// There is currently no GetBlobReference call that does not populate
+    /// the fields of an ICloudBlob.
+    /// </summary>
+    public sealed class AzureBlobFileSystem : IFileSystem
+    {
+        private readonly ICloudBlobClient _client;
+
+        public AzureBlobFileSystem(Uri baseUri)
+        {
+            _client = new AzureCloudBlobClient(baseUri);
+        }
+
+        public AzureBlobFileSystem(Uri baseUri, StorageCredentials storageCredentials)
+        {
+            _client = new AzureCloudBlobClient(baseUri, storageCredentials);
+        }
+
+        public AzureBlobFileSystem(StorageUri storageUri, StorageCredentials storageCredentials)
+        {
+            _client = new AzureCloudBlobClient(storageUri, storageCredentials);
+        }
+
+        public AzureBlobFileSystem(string connectionString)
+        {
+            _client = new AzureCloudBlobClient(connectionString);
+        }
+
+        [Inject]
+        private AzureBlobFileSystem(ICloudBlobClient client)
+        {
+            _client = client;
+        } 
+
+        /// <summary>
+        /// Not supported for Azure Blobs, will throw <see cref="NotSupportedException"/>.
+        /// </summary>
+        public Stream Open(Uri fileUri)
+        {
+            throw new NotSupportedException("Open is not supported for AzureBlobFileSystem.");
+        }
+
+        /// <summary>
+        /// Not supported for Azure Blobs, will throw <see cref="NotSupportedException"/>.
+        /// </summary>
+        public Stream Create(Uri fileUri)
+        {
+            throw new NotSupportedException("Open is not supported for AzureBlobFileSystem.");
+        }
+
+        /// <summary>
+        /// <see cref="IFileSystem.Delete"/>
+        /// </summary>
+        public void Delete(Uri fileUri)
+        {
+            var uriSplit = fileUri.AbsolutePath.Split(new[] { "/" }, StringSplitOptions.RemoveEmptyEntries);
+            if (uriSplit.Length <= 1)
+            {
+                throw new ArgumentOutOfRangeException(@"URI must be a file and cannot be a container.");
+            }
+
+            try
+            {
+                _client.GetBlobReferenceFromServer(fileUri).Delete();
+            }
+            catch (StorageException e)
+            {
+                if (e.RequestInformation.HttpStatusCode != (int) HttpStatusCode.NotFound)
+                {
+                    throw;
+                }
+            }
+        }
+
+        /// <summary>
+        /// <see cref="IFileSystem.Exists"/>
+        /// </summary>
+        public bool Exists(Uri fileUri)
+        {
+            try
+            {
+                return _client.GetBlobReferenceFromServer(fileUri).Exists();
+            }
+            catch (StorageException e)
+            {
+                if (e.RequestInformation.HttpStatusCode != (int) HttpStatusCode.NotFound)
+                {
+                    throw;
+                }
+
```

Comment:
Yes, I've tested it E2E in my other test class. It seems brittle and I don't really like it though, since we're essentially using Exceptions for control flow here. I'm wondering if we should switch to only support one type of blob to get around this...
See my comment above to @dkm2110 to why I'm making the ``getBlobReferenceFromServer`` call.